samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -8,144 +8,341 @@
 using namespace std;
 using namespace cv;
 
-string getGstDemuxPlugin(string container);
-string getGstAvDecodePlugin(string codec);
-
-int main(int argc, char *argv[])
+class GStreamerPipeline
 {
-    const string keys =
-          "{h help usage ? |           | print help messages   }"
-          "{p pipeline     |gst-default| pipeline name  (supported: 'gst-default', 'gst-vaapi', 'gst-libav', 'ffmpeg') }"
-          "{ct container   |mp4        | container name (supported: 'mp4', 'mov', 'avi', 'mkv') }"
-          "{cd codec       |h264       | codec name     (supported: 'h264', 'h265', 'mpeg2', 'mpeg4', 'mjpeg', 'vp8') }"
-          "{f file path    |           | path to file }"
-          "{fm fast        |           | fast measure fps }";
+ public:
+    // Preprocessing arguments command line
+    void init(int argc, char *argv[])
+    {
+        const string keys =
+              "{h help usage ? |           | print help messages   }"
+              "{m mode         |           | coding mode (supported: encode, decode) }"
+              "{p pipeline     |gst-default| pipeline name  (supported: 'default, gst-default', 'gst-vaapi', 'gst-libav', 'ffmpeg') }"
+              "{ct container   |mp4        | container name (supported: 'mp4', 'mov', 'avi', 'mkv') }"
+              "{cd codec       |h264       | codec name     (supported: 'h264', 'h265', 'mpeg2', 'mpeg4', 'mjpeg', 'vp8') }"
+              "{f file path    |           | path to file }"
+              "{vr resolution  |720p       | video resolution for encoding (supported: '720p', '1080p', '4k') }"
+              "{fps            |30         | fix frame per second for encoding (supported: fps > 0) }"
+              "{fm fast        |           | fast measure fps }";
+        CommandLineParser parser(argc, argv, keys);
+        parser.about("This program shows how to read a video file with GStreamer pipeline with OpenCV.");
 
-    CommandLineParser parser(argc, argv, keys);
+        if (parser.has("help"))
+        {
+            parser.printMessage();
+            exit(-1);
```

Comment:
Since you've wrapped all into a class it  should not influence whole program execution. Please change all `exit` calls to `return false` or similar statement.